experimental/dds/tree2/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -328,6 +335,176 @@ describe("SharedTree", () => {
+				const editable1 = view1.editableTree;
+
+				editable1.content = { [typeNameSymbol]: node.name, child: undefined };
+				containerRuntimeFactory.processAllMessages();
+
+				const tree2 = await factory.load(
+					dataStoreRuntime2,
+					"B",
+					{
+						deltaConnection: dataStoreRuntime2.createDeltaConnection(),
+						objectStorage: MockStorage.createFromSummary(
+							(await tree1.summarize()).summary,
+						),
+					},
+					factory.attributes,
+				);
+
+				containerRuntimeFactory.processAllMessages();
+				const incrementalSummaryContext = {
+					summarySequenceNumber: dataStoreRuntime1.deltaManager.lastSequenceNumber,
+
+					latestSummarySequenceNumber: dataStoreRuntime1.deltaManager.lastSequenceNumber,
+
+					summaryPath: "test",
+				};
+				const summaryTree = await tree2.summarize(
+					undefined,
+					undefined,
+					undefined,
+					incrementalSummaryContext,
+				);
+				containerRuntimeFactory.processAllMessages();
+				const indexes = summaryTree.summary.tree.indexes;
+				assert(indexes.type === SummaryType.Tree, "Indexes must be a tree");
+				const schemaBlob = indexes.tree.Schema;
+				assert(schemaBlob.type === SummaryType.Tree, "Blob Schema tree not created");
+				assert(
+					schemaBlob.tree.SchemaString.type === SummaryType.Handle,
+					"schemaString should be a handle",
+				);
+			});
+
+			it("on a client which uploaded a blob", async () => {
+				const provider = await TestTreeProvider.create(1, SummarizeType.onDemand);
+
+				await provider.ensureSynchronized();
+				const tree1 = provider.trees[0].schematizeInternal({
+					schema: stringSequenceRootSchema,
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: ["B"],
+				});
+
+				await provider.ensureSynchronized();
+				await provider.summarize();
+
+				tree1.editableTree.insertAt(0, ["A"]);
+
+				await provider.ensureSynchronized();
+				const { summaryTree } = await provider.summarize();
+				validateSchemaStringType(summaryTree, provider.trees[0].id, SummaryType.Handle);
+			});
+		});
+
+		describe("uploads new schema data", () => {
+			it("without incremental summary context", async () => {
+				const provider = await TestTreeProvider.create(1, SummarizeType.onDemand);
+				await provider.ensureSynchronized();
+				const summaryTree = await provider.trees[0].summarize();
+				const indexes = summaryTree.summary.tree.indexes;
+				assert(indexes.type === SummaryType.Tree, "Indexes must be a tree");
+				const schemaBlob = indexes.tree.Schema;
+				assert(schemaBlob.type === SummaryType.Tree, "Blob Schema tree not created");
+				assert(
+					schemaBlob.tree.SchemaString.type === SummaryType.Blob,
+					"schemaString should be a Blob",
+				);
+			});
+
+			it("when it has changed since the last summary", async () => {
+				const provider = await TestTreeProvider.create(1, SummarizeType.onDemand);
+
+				await provider.ensureSynchronized();
+				const tree1 = provider.trees[0].schematizeInternal({
```

Comment:
my addition of schema editing just merged (you'll have to change a few places where you call storedSchema.update to checkout.updateSchema). this also makes me think you could remove your use of schematizeInternal (or schematize) in any of the tests here that don't care about reading the tree content, similarly to how I used the untyped checkout in [this file](https://github.com/microsoft/FluidFramework/pull/18815/files#diff-1759b697b4c2f23e47152f7c6e39d42c08a3deb6cb2549281832acf23027f03f) in the "can be undone at the tip" test

Reply:
simplified test cases and removed use of schematizeInternal